mongo/mongocryptd.go

```diff
@@ -35,23 +35,27 @@ type mcryptClient struct {
 	spawnArgs   []string
 }
 
-func newMcryptClient(opts map[string]interface{}) (*mcryptClient, error) {
+func newMcryptClient(opts *options.AutoEncryptionOptions) (*mcryptClient, error) {
 	// create mcryptClient instance and spawn process if necessary
-	mc := &mcryptClient{}
-	if bypass, ok := opts["mongocryptdBypassSpawn"]; ok {
+	mc := &mcryptClient{
+		// if BypassAutoEncryption is specified, mongocryptd is not needed because it is only used for marking commands
+		// during encryption and is not used during decryption
+		bypassSpawn: opts.BypassAutoEncryption != nil && *opts.BypassAutoEncryption,
+	}
+	if bypass, ok := opts.ExtraOptions["mongocryptdBypassSpawn"]; ok {
```

Comment:
What's the expected behavior when: 
 
      *opts.BypassAutoEncryption == true
      opts.ExtraOptions["mongocryptdBypassSpawn"] == false

?  I would expect that `mc.bypassSpawn` should equal true but iiuc it will end up equal to false with this logic.

Reply:
Good catch. I cleaned up the logic for this to basically do
`bypassSpawn = mongocryptdBypassSpawn || bypassAutoEncryption`

Reply:
Much better.  Do you think the test coverage is sufficient given that it didn't catch this bug?

Reply:
It is not. As discussed offline, I'm going to add tests for the combinations of `mongocryptdByapssSpawn`/`bypassAutoEncryption` and make those table tests in the driver for convenience. The latest commit is a WIP for that. Will ping when I finish.

Reply:
@jyemin This is ready for another review from you.